cmd/fake-lepton/main.go

```diff
@@ -126,23 +129,41 @@ func connectToSocket(dbusService *service) error {
 }
 
 func sendCPTV(conn *net.UnixConn, file string) error {
-    mu.Lock()
-    defer mu.Unlock()
-    fullpath := path.Join(cptvDir, file)
-    if _, err := os.Stat(fullpath); err != nil {
-        fmt.Printf("%v does not exist\n", fullpath)
-        return err
-    }
-    log.Printf("sending raw frames of %v\n", fullpath)
-    r, err := cptv.NewFileReader(fullpath)
-    if err != nil {
-        return err
+    select {
+    case lockChannel <- struct{}{}:
```

Comment:
Would it make the code cleaner to read the file and then acquire the lock?   Is that possible in this case?

Reply:
I'm not sure how this would make the code cleaner? also then the file may be opened and help open for lockTimeout unnecessarily.